### PR TITLE
Don't run conflicting shortcodes in autogenerated descriptions

### PR DESCRIPTION
**Pull request by arnaudbroes**
Issue #1747

## Proposed changes

This fixes an issue where error WooCommerce can't display errors on the "My Account" page when the "Run shortcodes in autogenerated descriptions" setting is enabled in AIOSEOP.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Improves existing functionality
- Improves existing code

## Checklist

- [x] I've selected the appropriate branch (ie x.y rather than master).
- [x] I've created an appropriate title, descriptive of what the PR does.
- [x] Travis-ci runs with no errors.
- [x] I have added tests that prove my fix is effective/my feature works or have created an issue for it (if appropriate).

## Testing instructions

Enable "autogenerated descriptions" and "run shortcodes in autogenerated descriptions" in the General Settings menu, then navigate to the "My Account" page that WooCommerce generates and enter no or incorrect login details. WooCommerce should now be able to display error notifications again.

## Further comments

Specifically, this PR adds a new function (aioseop_contains_conflicting_shortcodes) in aioseop_functions.php where we can add more shortcodes in the future that are known to conflict with this setting.

Also adds a unit test for this new function.

### CODE REVIEW

**Comment by wpsmort** [comment]
@arnaudbroes This fix also needs to be applied when the Run Shortcodes in Descriptions option in the Social Meta module is enabled.  Currently that setting also breaks the error display in WooCommerce.

**Comment by arnaudbroes** [comment]
@wpsmort I've taken care of it.

**Comment by wpsmort** [comment]
@arnaudbroes I tested this and it no longer works when I have either Run shortcodes option enabled.  For example, if I enable Autogenerate Descriptions and Run Shortcodes In Autogenerated Descriptions then the WooCommerce error is not output.

**Comment by arnaudbroes** [comment]
@wpsmort I'm not able to reproduce that. Is it possible that you were testing this with another PR? If not, please give me a call so we can go over it.

**Comment by wpsmort** [comment]
@arnaudbroes I tested again and this works fine if you're using the Block Editor (Paragraph block, Custom HTML block, or Shortcode block).  It also works if you're using the Classic Editor.  However, it does not work if you're using the Classic block.  This is the block that is used when you convert from the classic editor to Gutenberg so it's definitely something we need to support.

Here's the steps to reproduce:

1. Edit the My account page and remove the Shortcode block from the page
2. Add the Classic block and enter the shortcode for the WooCommerce My Account page
3. Enable Run Shortcodes in AIO General Settings
4. Visit the My account page in an incognito window and login without a username and password

You should not see the warning message.

**Comment by arnaudbroes** [comment]
@wpsmort I believe I found a simple fix.

**Comment by wpsmort** [comment]
This is tested and is good.

**Comment by arnaudbroes** [comment]
@wpsmort please test this again as we've made a good improvement thanks to @EkoJR. It's now possible to run shortcodes after conflicting shortcodes.

Create a custom shortcode (just some text) using the Shortcoder plugin, then create a new Page with the [woocommerce_my_account] shortcode and then your own custom shortcode and see if the custom shortcode's content is used in the description.

**Comment by wpsmort** [comment]
This is tested and is good.

File: modules/aioseop_sitemap.php [hunk 1]
```diff
@@ -3763,7 +3763,7 @@ private function get_content_from_galleries( $content ) {
 					}
 					$code .= ']';
 
-					$gallery_content .= do_shortcode( $code );
+					$gallery_content = aioseop_do_shortcodes( $code );
```
[the remark this answers is not in the document]
Comment (arnaudbroes) [reply modules/aioseop_sitemap.php]: Correct.

**Pull request merged**